app/models/auction.rb

```diff
@@ -1,11 +1,29 @@
 class Auction < ActiveRecord::Base
   has_many :bids
   has_many :bidders, through: :bids
-  enum result: {not_applicable: 0, accepted: 1, rejected: 2}
-  enum type: {single_bid: 0, multi_bid: 1}
-  enum awardee_paid_status: {not_paid: 0, paid: 1}
-  enum published: {unpublished: 0, published: 1}
+  enum result: { not_applicable: 0, accepted: 1, rejected: 2 }
+  enum type: { single_bid: 0, multi_bid: 1 }
+  enum awardee_paid_status: { not_paid: 0, paid: 1 }
+  enum published: { unpublished: 0, published: 1 }
 
   # Disable STI
   self.inheritance_column = :__disabled
+
+  def winning_bid
+    if single_bid? && AuctionStatus.new(self).available?
+      nil
+    else
+      lowest_bid
+    end
+  end
```

Comment:
I don't think winning_bid should be in the AR model because it depends on the auction type and I'd rather keep basic methods only in here